packages/astro/src/core/build/vite-plugin-css.ts

```diff
@@ -23,47 +22,25 @@ interface PluginOptions {
 	target: 'client' | 'server';
 }
 
-// Arbitrary magic number, can change.
-const MAX_NAME_LENGTH = 70;
-
 export function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin[] {
 	const { internals, buildOptions } = options;
 	const { settings } = buildOptions;
 
 	let resolvedConfig: ResolvedConfig;
 
-	// Turn a page location into a name to be used for the CSS file.
-	function nameifyPage(id: string) {
-		let rel = relativeToSrcDir(settings.config, id);
-		// Remove pages, ex. blog/posts/something.astro
-		if (rel.startsWith('pages/')) {
-			rel = rel.slice(6);
-		}
-		// Remove extension, ex. blog/posts/something
-		const ext = npath.extname(rel);
-		const noext = rel.slice(0, rel.length - ext.length);
-		// Replace slashes with dashes, ex. blog-posts-something
-		const named = noext.replace(/\//g, '-');
-		return named;
-	}
-
 	function createNameForParentPages(id: string, ctx: { getModuleInfo: GetModuleInfo }): string {
 		const parents = Array.from(getTopLevelPages(id, ctx));
-		const proposedName = parents
-			.map(([page]) => nameifyPage(page.id))
-			.sort()
-			.join('-');
-
-		// We don't want absurdedly long chunk names, so if this is too long create a hashed version instead.
-		if (proposedName.length <= MAX_NAME_LENGTH) {
-			return proposedName;
-		}
+		const firstParentId = parents[0]?.[0].id || 'index';
+		const firstParentExt = npath.extname(firstParentId);
+		const firstParentName = firstParentId.slice(firstParentId.lastIndexOf('/') + 1, firstParentId.indexOf(firstParentExt));
```

Comment:
If `parents[0]?.[0].id` is `undefined` and fallbacks to `index`, I think `firstParentName` would be incorrect as it doesnt have an extension. I suppose we want `firstParentName` to be `index` in that case?

Or perhaps we can do something like:

```suggestion
		const firstParentId = parents[0]?.[0].id
		const firstParentName = firstParentId ? npath.basename(npath.dirname(firstParentId)) : 'index'
```